src/main/java/net/imglib2/algorithm/hough/HoughTransform.java

```diff
@@ -0,0 +1,258 @@
+	{
+		this( inputImage, voteSize, new ArrayImgFactory<>(), type, exService );
+	}
+
+	/**
+	 * Constructor for a HoughTransform with a specific ImageFactory. Use this
+	 * if you have something specific in mind as to how the vote data should be
+	 * stored.
+	 *
+	 * @param inputImage
+	 *            the image for the HoughTransform to operate over.
+	 * @param voteSize
+	 *            array indicating the size of the voteSpace. This is passed
+	 *            directly into ImageFactory to create a voteSpace image.
+	 * @param voteFactory
+	 *            the ImgFactory used to generate the voteSpace image.
+	 */
+	protected HoughTransform( final Img< T > inputImage, final long[] voteSize, final ImgFactory< S > voteFactory, final S type, final ExecutorService exService )
+	{
+		image = inputImage;
+		voteCursor = null;
+		pTime = 0;
+		voteSpace = voteFactory.create( voteSize, type );
+		peaks = null;
+		peakExclusion = new double[ voteSize.length ];
+		one = type.createVariable();
+		one.setOne();
+		Arrays.fill( peakExclusion, 0 );
+		this.exService = exService;
+	}
+
+	/**
+	 * Place a vote with a specific value.
+	 *
+	 * @param loc
+	 *            the integer array indicating the location where the vote is to
+	 *            be placed in voteSpace.
+	 * @param vote
+	 *            the value of the vote
+	 * @return whether the vote was successful. This here particular method
+	 *         should always return true.
+	 */
+	protected boolean placeVote( final int[] loc, final S vote )
+	{
+		if ( voteCursor == null )
+		{
+			voteCursor = voteSpace.randomAccess();
+		}
+
+		voteCursor.setPosition( loc );
+		voteCursor.get().add( vote );
+
+		return true;
+	}
+
+	/**
+	 * Place a vote of value 1.
+	 *
+	 * @param loc
+	 *            the integer array indicating the location where the vote is to
+	 *            be placed in voteSpace.
+	 * @return whether the vote was successful. This here particular method
+	 *         should always return true.
+	 */
+	protected boolean placeVote( final int[] loc )
+	{
+		return placeVote( loc, one );
+	}
+
+	/**
+	 * Returns an ArrayList of long arrays, representing the positions in the
+	 * vote space that correspond to peaks.
+	 *
+	 * @return an ArrayList of vote space peak locations.
+	 */
+	public ArrayList< long[] > getPeakList()
+	{
+		return peaks;
+	}
+
+	public boolean setExclusion( final double[] newExclusion )
+	{
+		if ( newExclusion.length >= peakExclusion.length )
+		{
+			System.arraycopy( newExclusion, 0, peakExclusion, 0, peakExclusion.length );
+			return true;
+		}
+		return false;
+	}
+
+	protected void setErrorMsg( final String msg )
+	{
+		errorMsg = msg;
+	}
+
+	/**
+	 * Pick vote space peaks with a {@link LocalExtrema}.
+	 *
+	 * @return whether peak picking was successful
+	 */
+	protected boolean pickPeaks( ExecutorService exService )
+	{
+		S maxValue = voteSpace.firstElement().copy();
+		maxValue.setReal( maxValue.getRealDouble() );
```

Comment:
This seems like a bug. The minimal peak value should probably be zero or one. What if `voteSpace.firstElement()` is the highest peak? Then, no other peak would be found.